### PR TITLE
Fix #321: PHP 7.4 EOL

### LINKED ISSUE #321: PHP 7.4 EOL
PHP 7.4 has been EOL for a while now, we should consider dropping support for it including removing it from composer.json, removing it from test workflows, and removing it from required status checks.

### PR DESCRIPTION
**Pull request by danepowell**
Fix #321

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     ],
     "license": "MIT",
     "require": {
-        "php": "^7.4 | ^8.0.2 | ^8.1 | ^8.2",
```
Comment (danepowell) [inline composer.json:12]: `8.0.2` was added in #225, I can find no justification for the oddly-specific constraint. Possibly it was just the latest version at the time. Anyway, the specificity of it conflicts with the config php definition below.

File: phpstan.neon [hunk 2]
```diff
@@ -10,8 +10,5 @@ parameters:
         -
             message: '%Parameter #1 \$client of method League\\OAuth2\\Client\\Provider\\AbstractProvider::setHttpClient\(\) expects GuzzleHttp\\ClientInterface, Eloquent\\Phony\\Mock\\Mock given.%'
             path: tests/Connector/ConnectorTest.php
-        -
-            # https://github.com/phpstan/phpstan/issues/1066
```
Comment (danepowell) [inline phpstan.neon:14]: No longer necessary due to psr/cache upgrade

**Comment by danepowell** [comment]
@typhonius I would suggest modifying the branch protections to _only_ require the new "check" job, which in turn will require all CI jobs to pass. I [prefer this pattern](https://github.com/acquia/cli/blob/eaae65d32be5991433457dd3cace7fb7d3214b90/.github/workflows/ci.yml#L96) because you don't have to keep updating branch protection settings every time a PHP version changes.

**Comment by danepowell** [comment]
@typhonius can you fix the branch protections so this can be merged (as suggested above), or give me admin access to do so? Thanks!

**Comment by typhonius** [comment]
> @typhonius can you fix the branch protections so this can be merged (as suggested above), or give me admin access to do so? Thanks!

Thanks for your work on this @danepowell.

The name `check` doesn't come up in branch protections when I search for it like it's shown in [this comment](https://github.com/orgs/community/discussions/26733#discussioncomment-3253155). Any ideas for what we need to change to get this working as expected?

image

**Comment by danepowell** [comment]
@typhonius try again. That UI only shows jobs that ran in the past week; I updated this PR which should trigger the 'check' job to show up.

**Comment by typhonius** [comment]
> @typhonius try again. That UI only shows jobs that ran in the past week; I updated this PR which should trigger the 'check' job to show up momentarily.

Awesome thanks for that, it should be set correctly now.

**Pull request merged**